scripts/ci/jobs/gke_version_compatibility_tests.py

```diff
@@ -65,23 +65,6 @@
     ]
 )
 
-# Support exception for latest central and sensor 3.74 as per
-# https://issues.redhat.com/browse/ROX-18223
-support_exceptions = [
-    ChartVersions(
-        central_version=latest_tag,
-        sensor_version=get_latest_helm_chart_version_for_specific_release(
-            "stackrox-secured-cluster-services", Release(major=3, minor=74)
-        ),
-    )
-]
-
-test_tuples.extend(
-    support_exception
-    for support_exception in support_exceptions
-    if support_exception not in test_tuples
-)
-
```

Comment:
This exception should stay in for now. Our compatibility tests only run on gke, and currently a new support exception for this is being negotiated. In the meantime we don't want to break compatibility.

Here's a slack thread discussing a potential extension of the support exception / new support exception: https://redhat-internal.slack.com/archives/C3W47R5LP/p1719433090785519

I would strongly suggest to keep testing against this version at least until this is resolved because it could save us major headaches if the extension is renewed and we somehow break compatibility with 3.74. Especially since going forward we are only going to test against  4.4 and newer aside from the exception. I have a PR here https://github.com/stackrox/stackrox/pull/11791 that I'm going to be merging once compatibility tests are no longer broken, and this will reduce the runtime of our compatibility tests a lot by testing against versions that have support according to our product lifecycles API instead of the last 3/2 central/sensor versions.

Reply:
For a bit of context, this test is failing now because we have turned off (and removed) upstream kernel driver builds. There's a strong argument that we should have disabled GKE driver builds for 3.74 a long while ago, but it has been ticking away which is why the test hasn't failed until now.

I'm not entirely sure about the status of the support exception with regard to Collector, but there's been some discussion in the team that runtime collection may not be a part of the extension. (cc @robbycochran.) But regardless, I'm not sure testing compatibility with 3.74 on GKE is representative of the support exception anyway, especially from Collector's perspective.

An alternative would be to disable collection for the 3.74 test, and not deploy any Collectors at all, but I'm not sure if that invalidates the intention of these compatibility tests or whether these tests are setup in such a way that we can disable collection for a single test (?)

Reply:
The intention of the compatibility tests is to catch API changes or other changes in communication channels that could break setups where central and sensor are not on the same version, in the case of this support exception the customer is running latest central against sensor on v3.74. My assumption would be that  breaking backwards compatibility in central / sensor communication would show in these tests, regardless of whether it's on GKE or OCP. 

Since we do not currently have compatibility tests outside of GKE, keeping this running sounds better to me than not testing against v3.74 at all. 

Can I ask for the reason for removing upstream kernel driver builds? If they have a high cost maybe it outweighs testing against 3.74, I'm unsure on this.

> whether these tests are setup in such a way that we can disable collection for a single test
It doesn't sound so easy to me. The way these tests work right now is by iterating over all test tuples and running the exact same tests for all of them

cc @msugakov maybe you can weigh in since you worked on compatibility tests with me before and are familiat with SUPPORTEX-21476

Reply:
> Can I ask for the reason for removing upstream kernel driver builds?

It's couple of reasons, first that we do not support eBPF on non-rhel kernels anymore (we've not built eBPF upstream for new versions of ACS since 4.2), and second that it has been a huge maintenance burden on the team, with frequent awkward failures and complex interaction between several components up and downstream.

I agree that we should still test compatibility with 3.74 but it's just not really possible to include collector with this cluster and ACS version. Unless of course we restore the kernel builds, but I'm extremely reluctant to do so if it is only to support this one test.

I've just pushed some changes that reinstate the 3.74 test, and attempted to switch the collection method between tests using a pre-test class, hopefully this is a reasonable middle ground where we get _some_ compatibility testing with Sensor on 3.74

Reply:
Thanks for tagging me in. I'll try to get to read the PR and the thread today. Indeed SUPPORTEX-21476 will likely continue but the terms are being defined <https://docs.google.com/document/d/1M5VqA9QC4t12Z0Uy4_ufp-0DNa2tuUEHPdy4VuXbrQ4/edit>.

Reply:
Version compatibility tests are important by themselves and in the context of this support exception. For the support exception they verify that we don't change Central in a way that it breaks 3.74 secured clusters.

You'll see in the doc I linked that we **may** offer `NO_COLLECTION` but the final call hasn't been made yet.
It's important at this point to find out whether compatibility tests will pass at all without collectors.

I don't see many other options how to implement this. Since 3.74 is offered with kernel module collection, we could create a dedicated OSCI job for testing 3.74 Secured Clusters with recent Central and run it on a cluster version that has a kernel module driver. This could be either pinned GKE version (with the caveat that it may go out of support by Google and stop being create-able) or a pinned OCP version (obviously, not 3.11 because modern Central may not work there; something 4.x).